src/ray/gcs/redis_context.cc

```diff
@@ -506,6 +564,14 @@ Status RedisContext::Connect(const std::string &address,
   // Ray has some restrictions for RedisDB. Validate it here.
   ValidateRedisDB(*this);
 
+  // try re-connect to redis primary node if current connection is sentinel
+  auto sentinel_status = ValidateAndConnectRedisSentinel(*this, password, enable_ssl);
+  if (!sentinel_status.IsTypeError()) {
+    // if not type error, this should be a sentinel redis. return the status code
+    // directly. continue otherwise.
+    return sentinel_status;
+  }
+
   // Find the true leader
```

Comment:
This is needed only for Redis Cluster.

I feel our high level code for `RedisContext::Connect` should be

```
if (redis_cluster) {
  ValidateRedisCluster() // make sure it only has 1 shard
  FindMasterUsingMOVED()
} else {
  // redis sentinel
  FindMasterUsingSentinel()
}
```

Reply:
thanks for the suggestion! refactored a little bit. so now the logic in `RedisContext::Connect` should be mostly untouched: added one validation of redis_sentienel and an else branch to handle sentinel.

Reply:
this should be ready for re-review.